packages/property-view/src/browser/resource-property-view/resource-property-view-label-provider.spec.ts

```diff
@@ -0,0 +1,157 @@
+import { ApplicationProps } from '@theia/application-package/lib/application-props';
+FrontendApplicationConfigProvider.set({
+    ...ApplicationProps.DEFAULT.frontend.config
+});
+
+import { Container } from '@theia/core/shared/inversify';
+import { TreeNode } from '@theia/core/lib/browser/tree/tree';
+import { ResourcePropertiesLabelProvider } from './resource-property-view-label-provider';
+import { LabelProvider, LabelProviderContribution } from '@theia/core/lib/browser/label-provider';
+import { ContributionProvider } from '@theia/core/lib/common';
+import { ResourcePropertiesCategoryNode, ResourcePropertiesItemNode } from './resource-property-view-tree-items';
+import { DEFAULT_INFO_ICON } from './resource-property-view-label-provider';
+
+disableJSDOM();
+
+let resourcePropertiesLabelProvider: ResourcePropertiesLabelProvider;
+
+describe('resource-property-view-label', () => {
+
+    before(() => {
+        disableJSDOM = enableJSDOM();
+        const container = new Container();
+        container.bind(ResourcePropertiesLabelProvider).toSelf().inSingletonScope();
+        container.bind(LabelProvider).toSelf().inSingletonScope();
+        container.bind<Partial<ContributionProvider<LabelProviderContribution>>>(ContributionProvider)
+            .toConstantValue({
+                getContributions: () => [],
+            })
+            .whenTargetNamed(LabelProviderContribution);
+        resourcePropertiesLabelProvider = container.get(ResourcePropertiesLabelProvider);
+    });
+
+    after(() => {
+        disableJSDOM();
+    });
+
+    const categoryNode: ResourcePropertiesCategoryNode = {
+        name: 'category',
+        id: '',
+        icon: 'iconCategory',
+        children: [],
+        parent: {
+            id: '',
+            parent: undefined,
+            children: []
+        },
+        categoryId: '',
+        expanded: false,
+        selected: false,
+    };
+
+    const itemNode: ResourcePropertiesItemNode = {
+        name: 'item',
+        id: '',
+        icon: 'iconItem',
+        selected: false,
+        parent: {
+            name: 'category',
+            id: '',
+            icon: '',
+            children: [],
+            parent: {
+                id: '',
+                parent: undefined,
+                children: []
+            },
+            categoryId: '',
+            expanded: false,
+            selected: false,
+        },
+        property: 'property'
+    };
+
+    describe('#canHandle', () => {
+        it('should handle a category node', () => {
+            expect(resourcePropertiesLabelProvider.canHandle(categoryNode)).to.be.greaterThan(0);
+        });
+
+        it('should handle an item node', () => {
+            expect(resourcePropertiesLabelProvider.canHandle(itemNode)).to.be.greaterThan(0);
+        });
+
+        it('should handle a tree node (not an item nor a category)', () => {
```

Comment:
?

```suggestion
        it('should not handle a tree node (not an item nor a category)', () => {
```

The `canHandle` here returns `0` meaning it should not be handled by this contribution.